ui/components/app/assets/asset-list/asset-list.tsx

```diff
@@ -104,32 +124,91 @@ const AssetList = ({ onClickAsset, showTokensLinks }: AssetListProps) => {
   // for EVM assets
   const shouldShowTokensLinks = showTokensLinks ?? isEvm;
 
-  const detectedTokensMultichain = useSelector(
-    getAllDetectedTokensForSelectedAddress,
+  const detectedTokensMultichain: {
+    [key: `0x${string}`]: Token[];
+  } = useSelector(getAllDetectedTokensForSelectedAddress);
+
+  const multichainDetectedTokensLength = Object.keys(
+    detectedTokensMultichain || {},
+  ).reduce(
+    (sum, key) => sum + detectedTokensMultichain[key as `0x${string}`].length,
+    0,
```

Comment:
Could we use `Object.values` instead of `Object.keys`? This way we don't need to index and type assert?

```ts
const length = Object.values(detectedTokensMultichain)
  .reduce((acc, tokens) => acc + tokens.length, 0)
```